Campaign page improvement

Lindharden opened the pull request:
This PR delivers a full redesign and functional overhaul of the Campaign pages:

1. Complete Campaign Page Remake

- The campaign page has been rebuilt from the ground up with a cleaner structure, improved UX, and more meaningful insights.

2. Campaigns Are Now Identified by UTM Campaign Name
- The page now focuses exclusively on individual campaigns grouped by their UTM campaign name.
- Only traffic associated with a defined UTM campaign name is shown.
- All other UTM sources without a campaign name are intentionally excluded.
- This makes the Campaign page a true campaign-level analytics view, without unrelated clutter.

3. Client-Side Pagination with SSR Bootstrapping
- The first page of campaign data is server-side rendered for fast initial load.
- Additional pages load client-side with optimized, incremental fetching.
- Query and pagination performance has been improved across the board.

4. Sparkline Insights in Campaign Cards
- Each campaign card contains a sparkline showing recent activity trends, offering quick visual insight without needing to expand the card.

5. Expanded Campaign Cards With Deep Insights
- Expanding a campaign card now reveals a much richer set of campaign analytics:
- Table overview with merged/optimized drilldown tables
- Audience profile showing top 3 browsers, OS, countries, and devices
- Donut chart replacing the previous pie chart for a cleaner breakdown view
- Additional UI minor improvements

flopper123 commented on dashboard/src/app/(protected)/dashboard/[dashboardId]/campaign/CampaignAudienceProfile.tsx:
```diff
@@ -0,0 +1,114 @@
+    case 'os':
+      return {
+        icon: <OSIcon name={rawLabel} className='h-3.5 w-3.5' />,
+        label: rawLabel,
```
Question: why does `DeviceIcon` have `type={rawLabel}` while the other icons have `name={rawLabel}`? 🤔

flopper123 commented on dashboard/messages/da.json:
```diff
@@ -1271,55 +1271,55 @@
+      "loading": "Indlæser kampagner...",
+      "campaignRow": {
+        "sessions": "{count, plural, one {# session} other {# sessioner}}",
+        "bounceRate": "Bounce Rate",
```
```suggestion
        "bounceRate": "Afvisningsprocent",
```

flopper123 commented on dashboard/messages/da.json:
```diff
@@ -1271,55 +1271,55 @@
+        "noData": "Ingen målgruppedata for denne kampagne i det valgte interval.",
+        "deviceTitle": "Enheder",
+        "browserTitle": "Browsere",
+        "osTitle": "Styresystem",
```
```suggestion
        "osTitle": "Operativsystem",
```

flopper123 commented on dashboard/messages/da.json:
```diff
@@ -1271,55 +1271,55 @@
+      },
+      "loading": "Indlæser kampagner...",
+      "campaignRow": {
+        "sessions": "{count, plural, one {# session} other {# sessioner}}",
```
nvm, misread

flopper123 commented on dashboard/src/app/(protected)/dashboard/[dashboardId]/campaign/CampaignPaginationControls.tsx:
```diff
@@ -0,0 +1,190 @@
+                      page === currentPage
+                        ? 'bg-primary text-primary-foreground'
+                        : 'text-muted-foreground hover:text-foreground hover:bg-muted/60'
+                    }`}
```
Nitpick: Use `cn` instead of string interpolation

flopper123 reviewed (changes requested):
9) On phone, if you have a lot of pages, the back button is hidden behind the sidebar button:

<img width="382" height="833" alt="image" src="https://github.com/user-attachments/assets/538c0f06-b47b-4e4c-8a6e-50fa44ec1b99" />

Lindharden commented on dashboard/src/app/(protected)/dashboard/[dashboardId]/campaign/CampaignList.tsx:
```diff
@@ -0,0 +1,317 @@
+      setPageIndex(result.pageIndex);
+      setPageSize(result.pageSize);
+    } catch {
+    } finally {
```
I've added a toast 👍

Lindharden commented:
> 1. The page does not respect updates to the time filters. Only after a full refresh (except the body of the accordion). If you for instance go a week back, refresh, then select "Last 24 hours", you should still see the empty state. Same the other way around:

Fixed

> 2. The dashboard filters doesn't change anything on the page:

Filters and comparison should not be enabled, I've disabled them 👍 

> 3. I'd suggest adding a nicer empty state, maybe something like the funnels empty state?

done 👍 

> 4. The layout on tablets is broken, I'd suggest you reuse the layout from the phones for tablets as well maybe?

> 5. (quibble) Something I noticed that I can't unsee again: You can see the border/border radius below the blue left-border:

Improved 👍 

> 6. On Italian, the accordion text gets multi-lined even though there's plenty of space to have it on one line. This also causes the text to be misaligned:

Fixed. I found a way to make it nicely auto expand the grid 👍 

> 7. It would maybe be nice to add a tiny border around the sparkline (much smaller than the one of the screenshot ofc), wdyt?
>    Also, the "dot" gets cut-off at the top right now:

Will not add this. I don't think a border is suitable here. The sparkline is part of the card and shouldn't be a card within the card. I have adjusted padding so the dot is no longer sliced 👍
